src/managers/ThreadManager.js

```diff
@@ -0,0 +1,231 @@
+   * * A ThreadChannel object
+   * * A Snowflake
+   * @typedef {ThreadChannel|Snowflake} ThreadChannelResolvable
+   */
+
+  /**
+   * Resolves a ThreadChannelResolvable to a Thread Channel object.
+   * @method resolve
+   * @memberof ThreadManager
+   * @instance
+   * @param {ThreadChannelResolvable} thread The ThreadChannel resolvable to resolve
+   * @returns {?ThreadChannel}
+   */
+
+  /**
+   * Resolves a ThreadChannelResolvable to a thread channel ID string.
+   * @method resolveID
+   * @memberof ThreadManager
+   * @instance
+   * @param {ThreadChannelResolvable} thread The ThreadChannel resolvable to resolve
+   * @returns {?Snowflake}
+   */
+
+  /**
+   * A number that is allowed to be the duration in minutes before a thread is automatically archived. This can be:
+   * * `60` (1 hour)
+   * * `1440` (1 day)
+   * * `4320` (3 days)
+   * * `10080` (7 days)
+   * @typedef {number} ThreadAutoArchiveDuration
+   */
+
+  /**
+   * Options for creating a thread
+   * @typedef {Object} ThreadCreateOptions
+   * @property {string} name The name of the new Thread
+   * @property {ThreadAutoArchiveDuration} autoArchiveDuration How long before the thread is automatically archived
+   * @property {MessageResolvable} [startMessage] The message to start a public or news thread from,
+   * creates a private thread if not provided
+   * @param {string} [reason] Reason for creating the thread
+   */
+
+  /**
+   * Creates a new thread in the channel.
+   * @param {ThreadCreateOptions} [options] Options
+   * @returns {Promise<ThreadChannel>}
+   * @example
+   * // Create a new public thread
+   * channel.threads
+   *   .create({
+   *     name: 'food-talk'
+   *     autoArchiveDuration: 60,
+   *     startMessage: channel.lastMessageID,
+   *     reason: 'Needed a separate thread for food',
+   *   })
+   *   .then(console.log)
+   *   .catch(console.error);
+   * // Create a new private thread
+   * channel.threads
+   *   .create({ name: 'mod-talk', autoArchiveDuration: 60, reason: 'Needed a separate thread for moderation' })
+   *   .then(console.log)
+   *   .catch(console.error);
+   */
+  async create({ name, autoArchiveDuration, startMessage, reason } = {}) {
+    let path = this.client.api.channels(this.channel.id);
+    if (startMessage) {
+      startMessage = this.channel.messages.resolveID(startMessage);
+      path = path.messages(startMessage);
+    }
+
+    const data = await path.threads.post({
+      data: {
+        name,
+        auto_archive_duration: autoArchiveDuration,
+      },
+      reason,
+    });
+
+    return this.client.actions.ThreadCreate.handle(data).thread;
+  }
+
+  /**
+   * The options for fetching multiple threads, the properties are mutually exclusive
+   * @typedef {Object} FetchThreadsOptions
+   * @property {FetchArchivedThreadOptions} [archived] The options used to fetch archived threads
+   * @property {boolean} [active] When true, fetches active threads. If archived is set, this is ignored!
```

Comment:
```suggestion
   * @property {boolean} [active] When true, fetches active threads.<warn>If `archived` is set, this is ignored!</warn>
```

Reply:
I'll try this out, not sure if this will inline where it seems, but worth a shot